### DISCUSSION TITLE
MUI X v6 DateTimePicker

### DISCUSSION THREAD

**Discussion by aintHuman** [lookfirst/mui-rff#998] [General]
has anyone got the DateTimePicker working with the new v6 mui release ?

https://next.mui.com/x/whats-new/

Specifically, use of the onError function can be used to render field validation errors, however, these errors are not populated back to the MUI-RFF form validation, and in fact the form can still be submitted. 

For example, an error function could be used like follows: (based off response  [HERE](https://stackoverflow.com/a/71999838/1834057) )

```
onError             = {(reason, value) => {
    if (reason) {
        // reason is the error , which it will be displayed as
        // default error message ,you can also  pass your own error
        // message depending on the error
        const errors = {
            minDate : `Below Min. Date (${minDate.format(DATE_FORMAT)})`,
            maxDate : "Max. Date Exceeded"
        }
        setCurrentError(`Please Check, ${errors[reason] || reason}`);
        setErrorDate(true);
    } else {
        setCurrentError(null);
        setErrorDate(false);
    }
}}
```

which can then be rendered by a `FormHelperText` object.

Also, incidentally, errors which are created by the form validation don't get rendered by the `DateTimePIcker,` in the same way as `DateTimePickers` for previous version, or standard `TextField`s etc....

I'd like to get this working, the user experience of entering data into the new V6 'fields' approach [(HERE)](https://next.mui.com/x/react-date-pickers/fields/) to entering data for pickers, is far superior IMO.

**Comment by lookfirst** [comment]
Hi there, thanks for bringing this up. I converted this to a discussion since it isn't really an issue. If we find things that could be an issue out of the discussion, we can file them as such.

If you can set up a reproducible example of how the error handling isn't working for you, I'd appreciate it. It would be easier for me to understand what you're bringing up.

It sounds like the fields approach is new, we don't support that yet. Happy to accept contributions on that for sure.
